commit 6f699e038a8b3a63af19e4b219ecdda98082fdd3 (refs/changes/77/5277/1)
Author: Rui Loura <rui@joyent.com>
Date:   2018-12-21T16:33:54+00:00 (10 months ago)
    
    Revert "TRITON-1053 CNAPI should not crash when someone calls `GET /servers/`"
    
    This reverts commit 22958c1b2d602fa1bc301a7f717dc9632dfce3c2.

diff --git a/lib/endpoints/index.js b/lib/endpoints/index.js
index 318b691..1ff4b46 100644
--- a/lib/endpoints/index.js
+++ b/lib/endpoints/index.js
@@ -77,18 +77,6 @@ function ensureConnectionTimeout(opts, req, res, next) {
     next();
 }
 
-function isUuid(candidate) {
-    // borrowed from assert-plus
-    /* JSSTYLED */
-    var UUID_REGEXP = /^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$/;
-
-    if (typeof (candidate) === 'string' && UUID_REGEXP.test(candidate)) {
-        return true;
-    }
-
-    return false;
-}
-
 
 /* BEGIN JSSTYLED */
 /**
@@ -115,14 +103,9 @@ function ensureServerRunning(opts, req, res, next) {
     });
 }
 
-function prepopulateServer(opts, req, res, next) {
-    assert.object(req, 'req');
-    assert.object(req.params, 'req.params');
 
-    if (!isUuid(req.params.server_uuid)) {
-        next(new restify.InvalidArgumentError('Invalid server_uuid'));
-        return;
-    }
+function prepopulateServer(opts, req, res, next) {
+    assert.string(req.params.server_uuid, 'server_uuid');
 
     ModelServer.get(
         req.params.server_uuid,
@@ -147,13 +130,6 @@ function prepopulateServer(opts, req, res, next) {
 
 function prepopulateVm(opts, req, res, next) {
     assert.string(req.params.uuid, 'uuid');
-    assert.object(req, 'req');
-    assert.object(req.params, 'req.params');
-
-    if (!isUuid(req.params.uuid)) {
-        next(new restify.InvalidArgumentError('Invalid vm_uuid'));
-        return;
-    }
 
     var uuid = req.params.uuid;
 
diff --git a/package.json b/package.json
index 5a87db4..63010f7 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.18.1",
+  "version": "1.18.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
